src/GovUk.Education.ExploreEducationStatistics.Common/Services/BlobStorageService.cs

```diff
@@ -640,7 +653,11 @@ private async Task<BlobClient> GetBlobClient(IBlobContainer containerName, strin
             var container = blobClient.GetContainerReference(
                 IsDevelopmentStorageAccount(blobClient) ? containerName.EmulatedName : containerName.Name);
 
-            await container.CreateIfNotExistsAsync();
+            await _storageInstanceCreationUtil.CreateInstanceIfNotExistsAsync(
+                _connectionString,
+                AzureStorageType.Blob,
+                containerName.Name,
```

Comment:
Ah ha, cool I see that, thanks.

Admittedly this is only an issue if the containers don't exist for a completely clean local setup, but that's the scenario where the creates are mainly going to be important. 🙂. 

If an app uses the `BlobStorageService` and anywhere gets both the 'private' and 'public' containers using the same connection string, i.e. locally, where the public container has a different name, then the public container won't be created if the private container is retrieved first, because as it stands the keys are the same.

See `GovUk.Education.ExploreEducationStatistics.Publisher.Services.PublishingService.PublishMethodologyFiles` in the call to `CopyDirectory` which is an example of where this would be a problem.

If we add a breakpoint just before that, and make sure `public-methodologies` doesn't already exist, it won't be created because the call to retrieve it uses the wrong key.

That code only happens to work without an exception because of this call directly before `CopyDirectory` which has created the public container first.

`            // Delete any existing blobs in public storage
            await _publicBlobStorageService.DeleteBlobs(
                containerName: PublicMethodologyFiles,
                directoryPath: directoryPath
            );
`

If this is commented out, then `CopyDirectory` fails with error `The specified container does not exist.`